File: python_backend/fenParser/getFen/get_piece_fen.py
from custom_except import *

def get_piece_fen(id):
    """getFen TASK 2.1, take the game id of a piece and convert it a fen id"""
    if id[0] == 'B':
        return id[1].lower()
    elif id[0] == 'W':
        return id[1]
    else:
        print('error occured  with game id or piece')
        return -1


if __name__ == "__main__":

    # white:
    print(get_piece_fen('WB2'))
    print(get_piece_fen('WQ1'))
    print(get_piece_fen('WK1'))
    print(get_piece_fen('WN1'))
    print(get_piece_fen('WR2'))
    print(get_piece_fen('WP3'))

    # black:
    print(get_piece_fen('BB2'))
    print(get_piece_fen('BQ1'))
    print(get_piece_fen('BK1'))
    print(get_piece_fen('BN1'))
    print(get_piece_fen('BR2'))
    print(get_piece_fen('BP3'))

File: app/src/game_logic/fenParser/getFen/getPieceFen.js


export function getPieceFen(id_) {
    /*getFen TASK 2.1, take the game id of a piece &&  convert it a fen id**/
    if (id_[0] === 'B') {
        return id_[1].toLowerCase()
    }
    else if (id_[0] === 'W') {
        return id_[1]
    }
    else {
        console.log('error occured  with game id || piece')
        return -1
    }
}

// module.exports = getPieceFen;

/** for node.js
if (require.main === module) {
    
    // white:
    console.log(getPieceFen('WB2'))
    console.log(getPieceFen('WQ1'))
    console.log(getPieceFen('WK1'))
    console.log(getPieceFen('WN1'))
    console.log(getPieceFen('WR2'))
    console.log(getPieceFen('WP3'))
    
    // black:
    console.log(getPieceFen('BB2'))
    console.log(getPieceFen('BQ1'))
    console.log(getPieceFen('BK1'))
    console.log(getPieceFen('BN1'))
    console.log(getPieceFen('BR2'))
    console.log(getPieceFen('BP3'))

}
*/

